collectors/ciscomeraki/collector.js

```diff
@@ -0,0 +1,251 @@
+
+    generateInitialStates(networkIds) {
+        const startTs = process.env.paws_collection_start_ts ?
+            process.env.paws_collection_start_ts :
+            moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+        const initialStates = networkIds.map(networkId => ({
+            networkId: networkId,
+            since: startTs,
+            until: endTs,
+            nextPage: null,
+            poll_interval_sec: parseInt(Math.floor(Math.random() * 30) + 1)
+        }));
+        return initialStates;
+    }
+
+    async handleEvent(event) {
+        let collector = this;
+        let context = this._invokeContext;
+        let parsedEvent = collector._parseEvent(event);
+
+        switch (parsedEvent.RequestType) {
+            case 'ScheduledEvent':
+                switch (parsedEvent.Type) {
+                    case 'SelfUpdate':
+                        await collector.handleUpdateStreamsFromNetworks();
+                        return collector.handleUpdate();
+                        break;
+                    default:
+                        super.handleEvent(event);
+                }
+            default:
+                super.handleEvent(event);
+        }
+    }
+
+    async handleUpdateStreamsFromNetworks() {
+        let collector = this;
+        try {
+            const payloadObj = utils.getOrgKeySecretEndPoint(collector.secret);
+            //get networks from api
+            let networks = await utils.getAllNetworks(payloadObj);
+            const keyValue = `${process.env.customer_id}/${collector._pawsCollectorType}/${collector.collector_id}/networks_${collector.collector_id}.json`;
+            let params = await utils.getS3ObjectParams(keyValue, undefined);
+            
+            //get networks json from s3 bucket
+            let networksFromS3 = await utils.fetchJsonFromS3Bucket(params.bucketName, params.key);
+            AlLogger.debug(`CMRI0000025 networks: ${JSON.stringify(networks)} networksFromS3 ${JSON.stringify(params)} ${JSON.stringify(networksFromS3)}`);
+            if (networks.length > 0 && Array.isArray(networksFromS3) && networksFromS3.length > 0) {
+                let differenceNetworks = utils.differenceOfNetworksArray(networks, networksFromS3);
+                AlLogger.debug(`CMRI0000024 Networks updated ${JSON.stringify(differenceNetworks)}`);
+
+                if (differenceNetworks.length > 0) {
+                    const initialStates = this.generateInitialStates(differenceNetworks);
+                    AlLogger.debug(`CMRI0000020: SQS message added ${JSON.stringify(initialStates)}`);
+                    collector._storeCollectionState({}, initialStates, this.pollInterval, async () => {
+                        await utils.uploadNetworksListInS3Bucket(keyValue, networks);
+                    });
+                } else {
+                    await utils.uploadNetworksListInS3Bucket(keyValue, networks);
```

Comment:
I think we don’t need to update json file here if there is no difference in network array.